### PR TITLE
Update README and SAM template descriptions

### LINKED ISSUE #5: Clean script does not delete the API Keys created
The clean script drops everything but the API Keys created. 

Is there particular reason for not doing that?

### PR DESCRIPTION
**Pull request by Trial-In-Error**
Closes #5

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -12,27 +12,45 @@ It also optionally supports subscription/unsubscription through a SaaS product o
...
 ```
 
 Followed by:
 
 ```bash
-sam deploy --template-file ./packaged.yaml --stack-name "dev-portal" --capabilities CAPABILITY_NAMED_IAM --parameter-overrides DevPortalSiteS3BucketName="dev-portal-static-assets" ArtifactsS3BucketName="dev-portal-artifacts"
+sam deploy --template-file ./packaged.yaml --stack-name "dev-portal" --capabilities CAPABILITY_NAMED_IAM --parameter-overrides DevPortalSiteS3BucketName="custom-prefix-dev-portal-static-assets" ArtifactsS3BucketName="custom-prefix-dev-portal-artifacts"
 ```
 
+Make sure to replace "custom-prefix" in the command above with some prefix that is globally unique, like your org name or username.
```
Comment (nitishdayal) [inline README.md]: I think it makes more sense to have this placed before the command.

File: README.md [hunk 2]
```diff
@@ -12,27 +12,45 @@ It also optionally supports subscription/unsubscription through a SaaS product o
...
+
 The command will exit when the stack creation is successful. If you'd like to watch it create in real-time, you can log into the cloudformation console.
 
-You can override any of the parameters in the template using the `--parameter-overrides key="value"` format. This will be necessary if you intend to deploy several instances of the developer portal.
+You can override any of the parameters in the template using the `--parameter-overrides key="value"` format. This will be necessary if you intend to deploy several instances of the developer portal. You can see a full list of overridable parameters in `cloudformation/template.yaml` under the `Parameters` section.
+
+### Populate the swagger catalog
```
Comment (nitishdayal) [inline README.md]: Swagger capitalized or no?

File: README.md [hunk 3]
```diff
@@ -48,11 +66,11 @@ Additional Resources:
 
 ### SAM Stack (template.yaml)
 
-Most components in the developer portal are managed by the SAM stack defined in template.yaml. New application components can be added to this template. Configuration values are fed to this template from the parameter overrides provided on the command line. If overrides are not provided, default values are used.
+All the components in the developer portal are managed by the SAM stack defined in template.yaml. New application components can be added to this template. Configuration values are fed to this template from the parameter overrides provided on the command line. If overrides are not provided, default values are used.
 
 ### UI (/app)
 
-The UI is a simple React application hosted in a public S3 bucket. The client side code communicates with the application backend via an API Gateway proxy API. For more information on updating the UI, see `./dev-portal/README.md`.
+The UI is a simple React application hosted in an S3 bucket. The assets are uploaded to the S3 bucket  The client side code communicates with the application backend via an API Gateway proxy API. For more information on updating the UI, see `./dev-portal/README.md`.
```
Comment (nitishdayal) [inline README.md]: Missing punctuation after 'The assets are uploaded to the S3 bucket'

File: README.md (hunk 2 above)
Comment (Trial-In-Error) [reply README.md]: Yep. Find-replace'd.

File: README.md (hunk 1 above)
Comment (Trial-In-Error) [reply README.md]: Agree!

File: README.md (hunk 3 above)
Comment (Trial-In-Error) [reply README.md]: Went ahead and finished the sentence...

File: README.md [hunk 4]
```diff
@@ -48,11 +64,11 @@ Additional Resources:
...
+All the components in the developer portal are managed by the SAM stack defined in template.yaml. New application components can be added to this template. Configuration values are fed to this template from the parameter overrides provided on the command line. If overrides are not provided, default values are used.
 
 ### UI (/app)
 
-The UI is a simple React application hosted in a public S3 bucket. The client side code communicates with the application backend via an API Gateway proxy API. For more information on updating the UI, see `./dev-portal/README.md`.
+The UI is a simple React application hosted in an S3 bucket. The assets are uploaded to the S3 bucket by the static-asset-uploader lambda function. The client side code communicates with the application backend via an API Gateway proxy API. For more information on updating the UI, see `./dev-portal/README.md`.
 
 ### Application Backend (/lambdas/backend)
```
[the remark this answers is not in the document]
Comment (Trial-In-Error) [reply README.md]: Agreed! I'll add it soon.

**Pull request merged**